plugin/src/main/java/org/opensearch/ml/rest/RestMLProfileAction.java

```diff
@@ -105,6 +162,173 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
         };
     }
 
+    private void sendDefaultProfileResponseWhenExceptionOccurred(XContentBuilder builder, RestChannel channel) {
+        try {
+            builder.endObject();
+            channel.sendResponse(new BytesRestResponse(RestStatus.OK, builder));
+        } catch (IOException e) {
+            log.error("IO exception occurred when building profile response!", e);
+            onFailure(channel, RestStatus.INTERNAL_SERVER_ERROR, "Failed to get ML node level profile", e);
+        }
+    }
+
+    private void buildModelDeploymentData(
```

Comment:
This method seems heavy. I did some refactor for ML task worker nodes and add target worker node in profile response, check https://github.com/opensearch-project/ml-commons/pull/656. So you can get target worker nodes directly from profile response, no need to query index..

For model name part, I think it's mainly for frontend. How about we just return model id in profile API, then frontend can query model index to get name. Another option: frontend cache model id and model name mapping, then no need to query model index every time.

Reply:
If so, I think this PR can be closed, and it seems no necessary to add a new node under the response, maybe we can add target worker nodes directly to `nodes` entry?